ifrs17/Test/QueriesScenarioDataTest.ipynb

```diff
@@ -0,0 +1,386 @@
+        {
+            "cell_type": "code",
+            "source": [
+                "queriedRawVars.SelectMany(x => x.Values).Sum().Should().Be(52.0);"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Scenario query, Workspace with no new scenario data, mismatching import format"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "await DataSource.DeleteAsync( await DataSource.Query<RawVariable>().ToArrayAsync() );",
+                "\nawait DataSource.UpdateAsync( bestEstimateRawVars.Concat(scenarioRawVars).ToArray() );",
+                "\nvar ws = Workspace.CreateNew();",
+                "\nws.Initialize(x => x.FromSource(DataSource).DisableInitialization<RawVariable>().DisableInitialization<IfrsVariable>());"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "var queriedRawVars = await ws.QueryPartitionedDataAsync<RawVariable,PartitionByReportingNodeAndPeriod>(DataSource, partitionScenarioMTUP.Id, partition.Id, ImportFormats.Actual);"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "queriedRawVars.SelectMany(x => x.Values).Sum().Should().Be(0);"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Scenario query, Workspace with no new scenario data and DataSource with scenario data"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "await DataSource.DeleteAsync( await DataSource.Query<RawVariable>().ToArrayAsync() );",
+                "\nawait DataSource.UpdateAsync( bestEstimateRawVars.Concat(scenarioRawVars).ToArray() );",
+                "\nvar ws = Workspace.CreateNew();",
+                "\nws.Initialize(x => x.FromSource(DataSource).DisableInitialization<RawVariable>().DisableInitialization<IfrsVariable>());"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "var queriedRawVars = await ws.QueryPartitionedDataAsync<RawVariable,PartitionByReportingNodeAndPeriod>(DataSource, partitionScenarioMTUP.Id, partition.Id, ImportFormats.Cashflow);"
```

Comment:
what if Scenario is YCU+10?
I expect, 
RawVariable to be BE from DataSource,
IfrsVariable (not BOP) to be BE from DataSource.
IfrsVariable (BOP) to be BE from DataSource period -1 even if there is a period -1 Scenario YCU+10 in the DataSource.

Reply:
The case of scenario parameters (such as YCU+10) where BE Vars are taken from the DataSource is covered in the test 2.5 of this notebook. 

What do you mean with the last two lines about IfrsVar (not) BOP?